### PR TITLE
Switch to the PyData sphinx theme

### PR DESCRIPTION
**Pull request by maximlt**
Keeping up with the series, I hope that this will build pretty well and that I can integrate it in the next release.

### CODE REVIEW

File: doc/_static/css/custom.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
+:root {
+    --pst-color-primary: 47,47,47;
+    --pst-color-link: 56,123,178;
+    --pst-color-link-hover: 254,203,56;
```
Comment (philippjfr) [inline doc/_static/css/custom.css:4]: These should be adapted to the HoloViews colors if they haven't already.

Comment (maximlt) [reply doc/_static/css/custom.css]: Ok done, the links are now in the same color as the font of the holoviews logo.

**Comment by maximlt** [comment]
https://github.com/holoviz/holoviews/pull/5163/commits/46ba8bbf20f8f8b84809c245173e62d6228b7da1 is a CSS fix that should be upstreamed to nbsite.

**Review by jbednar — APPROVED** [review]

**Comment by maximlt** [comment]
The failing tests are transient, let's merge!

**Pull request merged**